core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -366,7 +376,7 @@ object SparkSubmit extends CommandLineUtils {
         // If a python file is provided, add it to the child arguments and list of files to deploy.
         // Usage: PythonAppRunner <main python file> <extra python files> [app arguments]
         args.mainClass = "org.apache.spark.deploy.PythonRunner"
-        args.childArgs = ArrayBuffer(args.primaryResource, args.pyFiles) ++ args.childArgs
+        args.childArgs = ArrayBuffer(localPrimaryResource, localPyFiles) ++ args.childArgs
         if (clusterManager != YARN) {
           // The YARN backend distributes the primary file differently, so don't merge it.
           args.files = mergeFileLists(args.files, args.primaryResource)
```

Comment:
It is a behavior change here. I think we should use `localFiles` and `localPrimaryResource` instead of `args.files` and `args.primaryResource`.

Reply:
Why do you think it should use local files/primary resource here? `args.files` will be assigned to "spark.files" for non-yarn deploy, and Spark's fileserver will download them to local for all the executors, so it should be fine to keep as remote resources.

Reply:
Also here the changes restricted to client deploy mode, I'm not sure why it is related to standalone cluster mode?